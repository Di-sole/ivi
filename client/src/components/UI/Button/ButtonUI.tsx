import { ReactNode } from 'react';
import styles from './ButtonUI.module.scss';

interface ButtonUIProps {
    className: string;
    children: ReactNode;
    background: 'lightRed' | 'gray';
    variant: 'large' | 'medium' | 'small' | 'square';
    onClick?: () => void;
}

function ButtonUI({ className, children, background, variant, onClick }: ButtonUIProps) {
    return (
        <>
            <button className={[styles.button, className, 'button'].join(' ')} onClick={onClick}>
                {children}
            </button>

            <style jsx>
                {`
                    .button {
                        background: ${background === 'lightRed' ? '#ea003d' : /*background === 'gray' ? */ '#1f1b2e'};

                        padding: ${variant === 'large'
                            ? '10px 15px'
                            : variant === 'medium'
                            ? '10px 12px'
                            : variant === 'small'
                            ? '7px 11px'
                            : /*variant === 'square' ? */ '5px'};
                    }

                    .button:hover {
                        background: ${background === 'lightRed' ? '#e42e5f' : /*background === 'gray' ? */ '#2e2844'};
                    }
                `}
            </style>
        </>
    );
}

export default ButtonUI;
