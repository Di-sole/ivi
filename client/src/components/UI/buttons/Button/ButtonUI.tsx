import { ReactNode } from 'react';
import styles from './ButtonUI.module.scss';

interface ButtonUIProps {
    className: string;
    children: ReactNode;
    background: 'lightRed' | 'gray' | 'transparentWhite' | 'transparent';
    shape?: 'large' | 'medium' | 'small' | 'square' | 'none';
    onClick?: () => void;
}

function ButtonUI({ className, children, background, shape, onClick }: ButtonUIProps) {
    return (
        <>
            <button className={[styles.button, className, 'button'].join(' ')} onClick={onClick}>
                {children}
            </button>

            <style jsx>
                {`
                    .button {
                        background: ${background === 'lightRed'
                            ? '#ea003d'
                            : background === 'gray'
                            ? '#1f1b2e'
                            : background === 'transparentWhite'
                            ? 'rgba(255, 255, 255, 0.08)'
                            : 'transparent'};

                        padding: ${shape === 'large'
                            ? '10px 15px'
                            : shape === 'medium'
                            ? '10px 12px'
                            : shape === 'small'
                            ? '7px 11px'
                            : shape === 'square'
                            ? '5px'
                            : 'none'};
                    }

                    .button:hover {
                        background: ${background === 'lightRed'
                            ? '#e42e5f'
                            : background === 'gray'
                            ? '#2e2844'
                            : 'transparent'};
                    }
                `}
            </style>
        </>
    );
}

export default ButtonUI; /*'#1f1b2e'};*/
