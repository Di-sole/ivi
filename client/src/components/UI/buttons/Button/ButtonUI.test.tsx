import React from 'react';
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react';
import ButtonUI from './ButtonUI'

describe('ButtonUI SNAPSHOTS TESTS', () => {
    test('lightRed ButtonUI should not be changed', () => {
        // arrange
        render(
            <ButtonUI
                className='button'
                background='lightRed'
                variant='large'>
                Красная кнопка
            </ButtonUI>
        )
        const button = screen.getByText('Красная кнопка')

        // assert
        expect(button).toMatchSnapshot()
    })

    test('gray ButtonUI should not be changed', () => {
        // arrange
        render(
            <ButtonUI
                className='button'
                background='gray'
                variant='small'>
                Серая кнопка
            </ButtonUI>
        )
        const button = screen.getByText('Серая кнопка')

        // assert
        expect(button).toMatchSnapshot()
    })
})
