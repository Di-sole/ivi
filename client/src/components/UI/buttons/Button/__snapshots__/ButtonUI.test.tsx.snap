// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonUI SNAPSHOTS TESTS gray ButtonUI should not be changed 1`] = `
<button
  class="jsx-1446086431 button button button"
>
  Серая кнопка
</button>
`;

exports[`ButtonUI SNAPSHOTS TESTS lightRed ButtonUI should not be changed 1`] = `
<button
  class="jsx-2749339351 button button button"
>
  Красная кнопка
</button>
`;
